### ISSUE TITLE
Lazy load relations

### ISSUE THREAD

**Issue by dtkahl** [vuex-orm/vuex-orm#165]
Is there a way to lazy load relations on an existing entity? Currently the only way to retrieve relations is through query:

```js
let user = User.query().with('comments').find(42);
```

I'd like to be able to do this:

```js
let user = User.find(42);
let comments = user.lazy('comments');
```

In my case this would be helpful because I like to provide a queried model instance by a vuex getter but do not want to retrieve all possible relations by default but only if they needed.

**Comment by kiaking** [comment]
Yeah this is nice feature I think. I was thinking to add function base relationship resolver something like this.

```js
class User extends Model {
  static entity = 'users'

  static fields () {
    return {
      id: this.attr(null),
      name: this.string('')
    }
  }

  comments () {
    return this.hasMany(Comment, 'user_id')
  }
}

// And you can call it like...

const user = User.find(1)

user.comments() // Get comments.
```

Currently it's a bit hard to implement lazy load after fetching data since when `find` is called, the data is already being fetched. But I agree you don't always not want to eager load everything.

**Comment by siegy22** [comment]
Any updates on this? 😢

**Comment by kiaking** [comment]
@siegy22 Actually YES! The related discussion is going on at #163. That is the long thread but simply put, what we are trying to do is;

1. Create an index map of the relation and reference it instead of creating a whole new model instance. However this is planned to be implemented after version 1.0.0 because it might take long time.

If we implement above, you don't even need `lazy` method since you can reference the relationship anytime you want regardless of `with` query.

However, I guess this is going to take a while before we can release it.

Maybe a workaround could be to return query object and let `user` object execute `get` along with `with`...? Something like below.

```js
function getQuery () {
  return this.$store.getters['entities/posts/query']()
    .where('active', true)
}

function getUser () {
  return getQuery().with('comments').get()
}
```

**Comment by Glogang9** [comment]
git rm [file]

On Aug 22, 2018 7:08 AM, "Yves Siegrist" <notifications@github.com> wrote:

Any updates on this? 😢

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/vuex-orm/vuex-orm/issues/165#issuecomment-415009847>,
or mute the thread
